libbeat/processors/ratelimit/rate_limit_test.go

```diff
@@ -180,7 +181,7 @@ func TestRateLimit(t *testing.T) {
 func TestAllocs(t *testing.T) {
 	p, err := new(conf.MustNewConfigFrom(mapstr.M{
 		"limit": "100/s",
-	}))
+	}), logptest.NewTestingLogger(t, ""))
```

Comment:
The `*logp.Logger` returned by `logptest.NewTestingLogger` is not a zero-allocation logger. I think you will need to leave this and the call below in a state that reflects the original code, the test passes here is you replace the call with `logp.NewLogger("testing")` (also for the benchmark).

Reply:
This should be done for `BenchmarkRateLimit` as well.